### PR TITLE
BKTCLT-3 Upgrade to Node 16

### PR DESCRIPTION
**Pull request by dimitri-bourreau**
This PR aims to make bucketclient functional with Node v.16.

For now, the PR:

- Updates GitHub Actions so it uses Node 16 instead of 10.
- Updates some dependencies without breaking their API.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,8 +33,8 @@
     "mocha": ""
   },
   "dependencies": {
-    "agentkeepalive": "^4.1.3",
-    "arsenal": "scality/Arsenal#918a1d7",
+    "agentkeepalive": "^4.1.4",
+    "arsenal": "scality/Arsenal#918a1d7c899a7ecddcd1d6e7bbdd49b5becc5bd0",
```
Comment (miniscruff) [inline package.json]: There should be a tag for arsenal instead of using the git hash, if not we can make one

Comment (dimitri-bourreau) [reply package.json]: I guess there is no tag associated to this specific commit, `git show-ref --tags | grep 918a1d7` returns nothing. Would you agree then a tag should be created?

Comment (miniscruff) [reply package.json]: Hmm, there is probably a later tag to use that should be fine. Depends on what base branch is needed, could be 7x or 8x I don't know.

**Comment by miniscruff** [comment]
@bert-e reset

File: tests/unit/simple_tests.js [hunk 2]
```diff
@@ -90,17 +90,25 @@ Object.keys(env).forEach(key => {
     describe(`Unit tests with mockup ${key} server`, () => {
         let server;
         let client;
+        let sockets;
 
         beforeEach('start server', done => {
-            server = e.s(handler).listen(9000, done).on('error', done);
             client = e.c;
+            sockets = [];
+            server = e.s(handler)
```
Comment (miniscruff) [inline tests/unit/simple_tests.js]: @nicolas2bert adding the sockets destroy fixed it, but let me know if you think I should do something else

Comment (nicolas2bert) [reply tests/unit/simple_tests.js]: Nice catch!
We can force closing the open connections left by destroying the client HTTP(s) agent.

```
afterEach('stop server and close connections', () => { 
    server.close(); 
    e.c.agent.destroy(); 
});
```

**Comment by miniscruff** [comment]
@bert-e reset

**Review by nicolas2bert — APPROVED** [review]

**Review by naren-rajendran — APPROVED** [review]

**Pull request merged**